internal/db/update/schema.go

```diff
@@ -306,28 +337,36 @@ SELECT COUNT(name) FROM sqlite_master WHERE type = 'table' AND name = 'schemas'
 // trigger the associated Update value. It's required that the minimum key in
 // the map is 1, and if key N is present then N-1 is present too, with N>1
 // (i.e. there are no missing versions).
-func NewFromMap(versionsToUpdates map[int]schema.Update) *SchemaUpdate {
+func NewFromMap(internalUpdates, externalUpdates map[int]schema.Update) *SchemaUpdate {
 	// Collect all version keys.
-	versions := []int{}
-	for version := range versionsToUpdates {
-		versions = append(versions, version)
-	}
 
-	// Sort the versions,
-	sort.Ints(versions)
+	schemaUpdate := &SchemaUpdate{}
+	for i, update := range []map[int]schema.Update{internalUpdates, externalUpdates} {
+		versions := []int{}
+		for version := range update {
+			versions = append(versions, version)
+		}
+
+		// Sort the versions,
+		sort.Ints(versions)
```

Comment:
It concerns me that we use `map[int]schema.Update` for an ordered list of updates, given that maps are iterated in random order. I realise this is a historical carry over from LXD.

But I wonder if we can do better here and use a []schema.Update instead?

Reply:
As mentioned above, I've made all of the schema update maps into slices now.